cool-core/src/main/java/com/nus/cool/core/cohort/refactor/CohortProcessor.java

```diff
@@ -217,23 +219,55 @@ private Boolean checkMetaChunk(MetaChunkRS metaChunk) {
       String valueSchema = ft.getFilterSchema();
       MetaFieldRS valueMetaField = metaChunk.getMetaField(valueSchema);
       if (this.checkMetaField(valueMetaField, ft)) {
-        return true;
+        return false;
       }
     }
     return false;
   }
 
   /**
-   * Now is not implemented.
+   * Check if this dataChunk is required.
+   *
+   * @param chunk data chunk
+   * @return if this data chunk need to check
    */
-  public Boolean checkMetaField(MetaFieldRS metaField, Filter ft) {
+  public Boolean checkDataChunk(ChunkRS chunk) {
+
+    // 1. check birth selection
+    // if the metaChunk contains all birth filter's accept value, then the metaChunk
+    // is valid.
+    if (this.birthSelector.getBirthEvents() == null) {
+      return true;
+    }
+
+    // 1. check birth selection
+    for (EventSelection es : this.birthSelector.getBirthEvents()) {
+      for (Filter ft : es.getFilterList()) {
+        String checkedSchema = ft.getFilterSchema();
+        FieldRS dataField = chunk.getField(checkedSchema);
+        if (ft.getType().equals(FilterType.Range)) {
+          if (this.checkDataRangeFiled(dataField, ft)) {
+            return false;
+          }
+        }
+      }
+    }
+
+    // 2. check cohort selection
+    Filter cohortFilter = this.cohortSelector.getFilter();
+    String checkedSchema = cohortFilter.getFilterSchema();
+    FieldRS dataField = chunk.getField(checkedSchema);
+    if (cohortFilter.getType().equals(FilterType.Range)) {
+      return this.checkDataRangeFiled(dataField, cohortFilter);
+    }
+
```

Comment:
Is it finished ? It doesn't cover valueSelection and ageSelection, thus the process logic maybe lead to error. 
And according to the reference of this method, the return value is exactly the opposite. (True should be false, and false should be true)

Reply:
And there is no check for HashType filter